addons/stock/models/stock_move.py

```diff
@@ -204,7 +204,8 @@ def _compute_location_dest_id(self):
                 location_dest = move.picking_id.location_dest_id
             elif move.picking_type_id:
                 location_dest = move.picking_type_id.default_location_dest_id
-            if location_dest and move.location_final_id and move.location_final_id._child_of(location_dest):
+            if location_dest and move.location_final_id and \
+               (move.location_final_id._child_of(location_dest) or move.picking_type_id.code == 'outgoing'):
```

Comment:
It's not possible to skip it when changing company instead. I think some people could use strange push in customer. (to automaticaly create return for example)

Reply:
It's a bit tricky.
The compute is triggered at the creation of the move (since it's `precompute`), which happens in the initial `_run_pull()` that was triggered through the sale order line confirmation. Without this change, it will at first always find the destination from the picking type, which means:
- Single step -> Customers
- Multi steps -> Output

Now, the issue is that we need to change the dest only of the *last* step, which means:
- Single step -> ~~Customers~~ Inter-company transit
- Multi steps -> Output

Since it can happen on different steps, I can't really trust either `_run_pull` or `_run_push` to know where to make that change. To me, it makes more sense to have it contained in the compute.

I agree that this solution is less than ideal though, and maybe we could find a better condition for that. Of course, the real *good* solution would be to properly duplicate the delivery rules for inter-company transactions, but since we're in stable our hands are mostly tied on that part.